src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js

```diff
@@ -68,29 +74,131 @@ class MapViewVisualization extends Visualization {
+            const computed = this.prepareDataPointsHelper(parsedData,preparedDataPoints,accentColor);
+            latitude       = computed.latitude;
+            longitude      = computed.longitude;
+        } else {
+            if (parsedData.type === SCATTERPLOT_LAYER && parsedData.data.length) {
+                const computed = this.prepareDataPointsHelper(parsedData.data,preparedDataPoints,accentColor);
+                latitude       = computed.latitude;
+                longitude      = computed.longitude;
+            } else if (parsedData.type === GEO_MAP && parsedData.layers !== undefined) {
+                parsedData.layers.forEach(layer => {
+                    if (layer.type === SCATTERPLOT_LAYER) {
+                        let dataPoints = layer.data || [];
+                        const computed = this.prepareDataPointsHelper(dataPoints,preparedDataPoints,accentColor);
+                        latitude       = computed.latitude;
+                        longitude      = computed.longitude;
+                    } else {
+                        console.log("Currently unsupported deck.gl layer.")
+                    }
+                })
+            }
+        }
+        return {latitude,longitude,zoom}
+    }
+
+    /**
+     * Helper for prepareDataPoints, calculating also central point.
+     * @returns {{latitude: number, longitude: number}} - center.
+     */
+    prepareDataPointsHelper(dataPoints,preparedDataPoints,accentColor) {
+        let latitudes  = [];
+        let longitudes = [];
+        dataPoints.forEach(e => {
+            this.pushGeoPoint(preparedDataPoints,e,accentColor);
+            latitudes.push(e.latitude);
+            longitudes.push(e.longitude);
+        });
+        let latitude  = 0.0;
+        let longitude = 0.0;
+        if (latitudes.length && longitudes.length) {
+            let minLat = Math.min.apply(null,latitudes);
+            let maxLat = Math.max.apply(null,latitudes);
+            latitude   = (minLat + maxLat)/2;
+            let minLon = Math.min.apply(null,longitudes);
+            let maxLon = Math.max.apply(null,longitudes);
+            longitude  = (minLon + maxLon)/2;
+        }
+        return {latitude,longitude};
+    }
+
+    /**
+     * Pushes a new deck.gl-compatible point from GeoPoint.
+     * @param preparedDataPoints - List holding geoPoints.
+     * @param element - element to create new deck.gl point from.
+     * @param accentColor - accent color of IDE if element doesn't specify one.
+     */
+    pushGeoPoint(preparedDataPoints,element,accentColor) {
+        let radius = isNaN(element.radius) ? DEFAULT_RADIUS : element.radius;
+        preparedDataPoints.push({
+            position: [element.longitude,element.latitude],
+            color: element.color || accentColor,
+            radius: radius
         });
     }
 
+    /**
+     * Sets size of the visualization.
+     * @param size - new size.
```

Comment:
this is not a proper doc. Just write such info, that I odnt need to look at implementation. Size is a list of two numbers containing widht and height repsectvively, right ?